src/main/scala/chisel3/package.scala

```diff
@@ -32,6 +32,10 @@ package object chisel3 {    // scalastyle:ignore package.object.name
     @deprecated("Wire(t, init) is deprecated, use WireInit(t, init) instead", "chisel3")
     def apply[T <: Data](t: T, init: T)(implicit compileOptions: CompileOptions): T =
       chisel3.core.WireInit(t, init)
+
+    @deprecated("Wire(t, init) is deprecated, use WireInit(t, init) instead", "chisel3")
```

Comment:
I don't think it's necessary to maintain compatibility in this case - DontCare came after this so I doubt there's legacy code where this is important.

Reply:
That's fair, I agree with not adding this "compatibility"

Reply:
At the same time, this can help reduce surprise for people looking at legacy examples of code.